### PR TITLE
[PD-2611] Multiple Communication Records

### PR DESCRIPTION
**Pull request by darrint**
This fixes a number of bugs regarding writing records to the database.

## Test

1. Pick a partner and contact.
2. Add an additional contact.
3. Add tags to the communication record. Set a date that is recent.
3. Go to PRM and verify that 2 communication records have been created and all tags are present on all records.

### CODE REVIEW

**Comment by mklauber** [comment]
:+1:

File: mypartners/models.py [hunk 1]
```diff
@@ -703,6 +703,25 @@ def save(self, *args, **kwargs):
 
         super(ContactRecord, self).save(*args, **kwargs)
 
+    def recreate(self):
```
Comment (tdphillips) [inline mypartners/models.py]: I know the documentation explains this, but the premise of this function feels weird to me.

File: mypartners/views.py [hunk 2]
```diff
@@ -2242,6 +2246,14 @@ def all_forms_iter():
             contact = contact_result['contact'].instance
         else:
             contact = contact_result['instance']
+
+        # on subsequent passes, make new communication records for each contact
+        if i > 0:
+            communication_record.recreate()
+            attach_new_tags(
```
Comment (tdphillips) [inline mypartners/views.py]: Given tags are added before recreate() is called, is this necessary?

File: mypartners/models.py (hunk 1 above)
Comment (darrint) [inline mypartners/models.py]: I don't think there's another way to write it. The technique is supposed to be set `pk` to `None` and `save()`. So you're stuck with mutating the original object in place.

File: mypartners/views.py (hunk 2 above)
Comment (darrint) [inline mypartners/views.py]: You were correct here. I'll put the fix for this in my next PR.

**Pull request merged**